t/t6025-merge-symlinks.sh

```diff
@@ -17,14 +16,14 @@ test_expect_success 'setup' '
     git commit -m initial &&
     git branch b-symlink &&
     git branch b-file &&
-    l="$(printf file | 
-    git hash-object -t blob -w --stdin)" &&
+    l="$(printf >file 
```

Comment:
See this is the problem with sub-shells. We can't capture error out of these. `printf >file` makes no sense (printf requires atleast one argument"). Which returns error and variable `l` is now empty. Let's not touch sub-shells at all.

Reply:
you can do something similar to:
```suggestion
	l="$(printf file >name && git hash-object -w name)"
```

PS: you can use this trick in all other sub-shells too.